packages/analytics-js/src/app/RudderAnalytics.ts

```diff
@@ -165,62 +165,102 @@ class RudderAnalytics implements IRudderAnalytics<IAnalytics> {
    * @returns
    */
   trackPageLifecycleEvents(loadOptions?: Partial<LoadOptions>) {
-    const { trackPageLifecycle, useBeacon } = loadOptions ?? {};
+    const { autoTrack, useBeacon } = loadOptions ?? {};
+    const {
+      enabled: autoTrackEnabled = false,
+      options: autoTrackOptions = {},
+      pageLifecycle,
+    } = autoTrack ?? {};
     const {
       events = [PageLifecycleEvents.LOADED, PageLifecycleEvents.UNLOADED],
-      enabled = false,
-      options = {},
-    } = trackPageLifecycle ?? {};
+      enabled = autoTrackEnabled,
+      options = autoTrackOptions,
+    } = pageLifecycle ?? {};
 
-    const visitId = state.pageLifecycle.visitId.value;
-    const pageLoadedTimestamp = state.pageLifecycle.pageLoadedTimestamp.value as number;
+    const preloadedEventsArray = this.getPreloadedEvents();
 
-    if (!enabled) {
-      return;
+    if (enabled) {
+      this.trackPageLoadedEvent(events, options, preloadedEventsArray);
+      this.setupPageUnloadTracking(events, useBeacon, options);
     }
 
-    const preloadedEventsArray = this.getPreloadedEvents();
+    // The array will be mutated in the below method
+    promotePreloadedConsentEventsToTop(preloadedEventsArray);
 
-    // track page loaded event
+    setExposedGlobal(GLOBAL_PRELOAD_BUFFER, clone(preloadedEventsArray));
```

Comment:
Let's see if we could move them out of this function. They seem to be misfit here.

Reply:
Addressed